smurfs/src/components/App.js

```diff
@@ -61,7 +76,26 @@ class App extends Component {
           <input type="text" name="name" value={this.state.newSmurf.name} onChange={this.changeHandler} placeholder="name" />
           <input type="number" name="age" value={this.state.newSmurf.age} onChange={this.changeHandler} placeholder="age" />
           <input type="text" name="height" value={this.state.newSmurf.height} onChange={this.changeHandler} placeholder="height" />
-          <button>Add Known Smurf</button>
+          {!this.state.editingSmurf ? (
+            <button onClick={this.addSmurf}>Add Known Smurf</button>
+          ) : (
+              <button
+                onClick={e => {
+                  this.props.editSmurf(this.state.newSmurf);
+                  this.setState({
+                    newSmurf: {
```

Comment:
Handling the state clearing when you call the edit -- in-line, is so organized. I never do this, but I love it.